File: edition/2017/java/Sorts/bubble_sort.java
/**
 * Created by renu.yadav on 20/10/17.
 */
public class bubble_sort {
    /**
     * bubble sort , compares adjacent values and swap them if needed,
     * after every  ith iteration ,ith biggest element would be in place
     * takes O(n^2) and an inplace algorithm with no extra space.
     * @param arr
     * @return
     */
    //
    public int[] bubbleSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < arr.length - i - 1; j++) {
                if (arr[j + 1] < arr[j]) {
                    // swap elements
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
        return arr;
    }
}

File: csharp/BubbleSort.cs
using System;

namespace BubbleSort {
	class Program {
		static void swap(ref int a, ref int b) {
			int c = a;
				a = b;
				b = c;
		}

		static void sort(int[] arr) {
			int length = arr.Length;
			for (int i = 0; i < length; i++) {
				for (int j = 0; j < length-i-1; j++) {
					if (arr[i] < arr[j]) {
						swap(ref arr[i], ref arr[j]);
					}
				}
			}
		}

		static void Main() {
			int[] arr = { 9, 4, 5, 6, 5, 3, 2, 10, 200, 1 };
			sort(arr);
			for (int i = 0; i < arr.Length; i++) {
				Console.Write($"{arr[i]} ");
			}
			Console.WriteLine();
		}
	}
}